src/neptune_scale/sync/sync_process.py

```diff
@@ -464,7 +343,32 @@ def _raise_exception(status_code: int) -> None:
         raise NeptuneUnexpectedResponseError()
 
 
-class StatusTrackingThread(Daemon, WithResources):
+def _merge_operations(operations: list[Operation]) -> tuple[UpdateRunSnapshots, SequenceId, datetime.datetime]:
+    snapshot_batch = UpdateRunSnapshots(snapshots=[op.operation for op in operations])  # type: ignore
+    return snapshot_batch, operations[-1].sequence_id, operations[-1].ts
+
+
+def _stream_operations(
+    operations_repository: OperationsRepository,
+    run_id: str,
+    project: str,
+    max_batch_size: int = 15 * 1024 * 1024,
+) -> Generator[tuple[RunOperation, SequenceId, datetime.datetime], None, None]:
+    while operations := operations_repository.get_operations(up_to_bytes=max_batch_size):
```

Comment:
Note to self: the repository needs to have some property describing max single operation size and
1. enforce validation on insert
2. make sure `up_to_bytes` is not lower than this value.

As it is now, we risk returning no operations if the parameters in the main and sync process don't play well together.

Reply:
done